modules/rosa-classic-cluster-terraform-file-creation.adoc

```diff
@@ -0,0 +1,297 @@
+  type        = bool
+  description = "Multi AZ Cluster for High Availability"
+  default     = true
+}
+
+variable "machine_type" {
+  description = "The AWS instance type used for your default worker pool"
+  type        = string
+  default     = "m5.xlarge"
+}
+
+variable "worker_node_replicas" {
+  default     = 3
+  description = "Number of worker nodes to provision. Single zone clusters need at least 2 nodes, multizone clusters need at least 3 nodes"
+  type        = number
+}
+
+
+variable "autoscaling_enabled" {
+  description = "Enables autoscaling. This variable requires you to set a maximum and minimum replicas range using the 'max_replicas' and 'min_replicas' variables."
+  type        = string
+  default     = "false"
+}
+
+variable "min_replicas" {
+  description = "The minimum number of replicas for autoscaling."
+  type        = number
+  default     = 3
+}
+
+variable "max_replicas" {
+  description = "The maximum number of replicas not exceeded by the autoscaling functionality."
+  type        = number
+  default     = 3
+}
+
+variable "proxy" {
+  default     = null
+  description = "cluster-wide HTTP or HTTPS proxy settings"
+  type = object({
+    http_proxy              = string           # required  http proxy
+    https_proxy             = string           # required  https proxy
+    additional_trust_bundle = optional(string) # a string contains contains a PEM-encoded X.509 certificate bundle that will be added to the nodes' trusted certificate store.
+    no_proxy                = optional(string) # no proxy
+  })
+}
+
+variable "aws_subnet_ids" {
+  type        = list(any)
+  description = "A list of either the public or public + private subnet IDs to use for the cluster blocks to use for the cluster"
+  default     = ["subnet-01234567890abcdef", "subnet-01234567890abcdef", "subnet-01234567890abcdef"]
+}
+
+variable "private_cluster" {
+  type        = bool
+  description = "If you want to create a private cluster, set this value to 'true.' If you want a publicly available cluster, set this value to 'false.'"
+}
+
+#VPC Info
+variable "vpc_name" {
+  type        = string
+  description = "VPC Name"
+  default     = "mobb-tf-vpc"
+}
+
+variable "vpc_cidr_block" {
+  type        = string
+  description = "value of the CIDR block to use for the VPC"
+  default     = "10.0.0.0/16"
+}
+
+variable "private_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the private subnets"
+  default     = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+}
+
+variable "public_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the public subnets"
+  default     = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+}
+
+variable "single_nat_gateway" {
+  type        = bool
+  description = "Single NAT or per NAT for subnet"
+  default     = false
+}
+
+#AWS Info
+variable "aws_region" {
+  type    = string
+  default = "us-east-2"
+}
+
+variable "admin_username" {
```

Comment:
This currently does not work, see https://redhat-internal.slack.com/archives/C03RNFXM72M/p1714734117012479